internal/engine/engine.go

```diff
@@ -412,6 +412,13 @@ func untar(ctx context.Context, dst string, r io.Reader) error {
 
 		// if it's a file create it
 		case tar.TypeReg:
+			// If the file's parent dir doesn't exist, create it.
+			dirname := filepath.Dir(target)
+			if _, err := os.Stat(dirname); err != nil {
```

Comment:
This is now happening in every branch. Perhaps it should be outside the switch?

Reply:
I see what you mean, but I think it has to happen in the switch because it conditionally happens in some of the cases (e.g. Symlinks), and must happen (though we could probably make it work) in others (e.g. Directories)

Reply:
It doesn't conditionally happen though. It happens in every case. The only thing that changes is that `target` is either the real thing in the case of tar.TypeDir, or just the Dir portion of the path.

Reply:
Synced up with @bcrochet but to document it here: Symlinks are skipped if they refer to a path outside of the tarball root once fully resolved (e.g. by way of `../../../../`). We skip them without writing anything. For this reason, we cannot write before the case statement.